### PR TITLE
(ts) fewer runtime globals (step 1)

### PR DESCRIPTION
**Pull request by sjmiles**
Step 2 
- get rid of remaining `staticRuntime` objects (runtime.ts, planner.ts)
- attempt to convert some globals like RamDiskStorageDriverProvider into runtime instances
- attempt to convert Planner statics into instance methods (or require `runtime` parameter)

### CODE REVIEW

**Review by mmandlis — APPROVED** [review]

File: src/runtime/runtime.ts [hunk 1]
```diff
@@ -307,41 +261,15 @@ export class Runtime {
...
-      storageManager
-    };
-    return serialization ? Arc.deserialize(params) : new Arc(params);
-  }
-
   // static interface for the default runtime environment
 
   static async parse(content: string, options?): Promise<Manifest> {
-    return this.getRuntime().parse(content, options);
+    return staticRuntime.parse(content, options);
```
Comment (shans) [inline src/runtime/runtime.ts:267]: will we get rid of these eventually too?

**Review by shans — APPROVED** [review]
Dude this is super nice!

File: src/runtime/runtime.ts (hunk 1 above)
Comment (sjmiles) [reply src/runtime/runtime.ts:267]: Yes, ASAP.

Comment (jblebrun) [reply src/runtime/runtime.ts:267]: Somewhat related to this: I was digging into the internal checks failure, this seems to fix it: cl/345574453

I couldn't find other uses of Runtime.parse/Runtime.parseFile, but might not have checked all of the right places.

**Review by jblebrun — APPROVED** [review]

**Pull request closed without merging**